### PR TITLE
rewrite using altair

### PR DESCRIPTION
**Pull request by Casyfill**
the very first passing implementation, mainly

- added `altair` as a dependency
- dropped `axis.py`
- dropped  some altair-specific tests (perhaps should drop more)
- every `plot` function now returns a "non-interactive" altair chart, therefore I dropped `interactive` and `ax` as arguements

### Some remarks
- definitely should try "reenact" this in the notebook https://pandas.pydata.org/pandas-docs/stable/visualization.html
- perhaps it's worth considering to add the support for `repeat` in api - for example, plotting the groupby object,  or something alike

### CODE REVIEW

**Comment by domoritz** [comment]
🎉

**Comment by domoritz** [comment]
When this is merged, should pdvega move to the altair org?

**Comment by Casyfill** [comment]
I'd appreciate any comments on the code, there is definitely room for improvements.
by the way, is there a reason `flake8` is not in the `test_requirements`?

**Comment by domoritz** [comment]
👍 to screen shot 2018-08-11 at 14 42 25

**Comment by Casyfill** [comment]
well those 6450 lines are essentially Altair :-P

**Comment by Casyfill** [comment]
ha, and those andrews curves in the notebook didn't help the count :-/

**Comment by jakevdp** [comment]
This looks great! I'll try to do a detailed review this week.

And yes, it would be worth moving this repo to the altair-viz org at some point.

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,7 @@ __pycache__/
 *.py[cod]
 *$py.class
 
-# C extensions
+# C extnsions
```
Comment (domoritz) [inline .gitignore]: typo?

Comment (domoritz) [inline .gitignore]: typo?

File: pdvega/plotting.py [hunk 2]
```diff
@@ -35,78 +33,80 @@ def scatter_matrix(frame, c=None, s=None, figsize=None, dpi=72.0, **kwds):
...
+                    "resolve": "global",
+                    "bind": "scales",
+                    "translate": "[mousedown[!event.shiftKey], window:mouseup] > window:mousemove!",
+                    "zoom": "wheel![!event.shiftKey]",
+                },
+            },
+            "encoding": {
+                "x": {"field": {"repeat": "column"}, "type": "quantitative"},
+                "y": {"field": {"repeat": "row"}, "type": "quantitative"},
+                "color": {"condition": {"selection": "brush"}, "value": "grey"},
```
Comment (jakevdp) [inline pdvega/plotting.py]: Any reason this isn't being built in Altair?

**Comment by jakevdp** [comment]
One over-arching comment: one thing this removes is the axes object, which I had created so that users could pass it around the same way they do in pandas; e.g.
```
ax = df.vgplot.line(...)
df.vgplot.scatter(..., ax=ax)
```
Can we make that work now that the return value is an Altair chart rather than an axes object?

**Comment by Casyfill** [comment]
re axes - we sure can; I personally feel it's an overkill, given same can be achieved with `+`, but let me add those.

I wonder if, given that many similar arguments, it makes sense to produce a base plot function, and inherit from it, (or advance the `_plot` function) - but it will also decrease the readability of the code, of course

**Comment by Casyfill** [comment]
will add testing for passing ax argument

File: pdvega/plotting.py (hunk 2 above)
Comment (jakevdp) [reply pdvega/plotting.py]: OK, I'm fine leaving this as-is for now.

**Comment by Casyfill** [comment]
added ax tests;
btw, It seems that `COMMON_ARGS` in `tests_core_common.py` is not used and wasn't used in original version, shall I drop it?

**Comment by jakevdp** [comment]
> shall I drop it

yes feel free.

File: pdvega/plotting.py (hunk 2 above)
Comment (Casyfill) [reply pdvega/plotting.py]: great!  I have this and a couple of other advancements (like symbol mapping for scatterplots, etc)
on my list for the next PR

**Comment by Casyfill** [comment]
let me know if there is anything else I can do before merging. 
As I mentioned, I have a list of features I'd like to add/improve, but perhaps it's safer to merge / check for bugs first

**Comment by jakevdp** [comment]
Yeah, I think it would be a good idea to merge and then iterate from there.

Thanks for getting this going!

**Comment by domoritz** [comment]
Wohoo

**Comment by Casyfill** [comment]
Hey @jakevdp, my team and I are still testing this build in the wild, and will most likely add another PR at some point soon, but do you think it is good enough to bump version as is on the PYPI?

**Comment by domoritz** [comment]
And move the repo over to altair-vis?

**Comment by jakevdp** [comment]
I think we should wait on a release... in particular, much of the documentation at http://jakevdp.github.io/pdvega will need to be updated. I don't want to have a release on PyPI that is largely incompatible with published documentation.

**Comment by Casyfill** [comment]
makes sense! 
docs are built from this repo, right? I changed most of the docstrings, can try to update documentation as well, then. 


side Q - in my view, matching matplotlib/pandas api is primarily useful for overriding `.plot` with `vgplot` with little to no other changes; in that sense, do you think it might be useful to add `figsize` argument? I noticed this is my main pain point for now

**Comment by jakevdp** [comment]
Sure, sounds good.

**Comment by jakevdp** [comment]
I think parity with ``df.plot`` is the eventual goal, so any step toward that is a good one!

**Pull request merged**